### ISSUE TITLE
Python 3.10 - scikit-learn 1.4+

### ISSUE THREAD

**Issue by dariopasquali** [bbonik/facial-expression-analysis#2]
Hi, I'm Dario Pasquali from the Italian Institute of Technology.
I'm using your code to detect the arousal and valence of people interacting with the humanoid robot iCub (don't worry, the citation is coming :smile: ).

Recently, I had to update the robot operating system to Ubuntu 22, and sadly I cannot use Python 3.6 and scikit-learn 0.23 anymore.
Two main reasons:
- Scikit-learn moved to the pytoml-based installation
- the PLS regressor class is different, hence your trained model cannot be used anymore with newer versions of scikit-learn.

I could search for other solution, but I really like yours as it is easy, fast and accurate enough for my application.
So, would it be possible for you to update your trained model to Python3.10 and scikit-learn 1.4+ ?

Thank you!
Dario

**Comment by bbonik** [comment]
Hi Dario,

Thank you for your interest in this project.
I know about the issue with the old SkLearn version.

Unfortunately I don't have the bandwidth to touch this project anymore, at
least not in the near future.
I will keep it in mind and if I find some time, I will try to update it,
but not anytime soon.

Best regards,
V

**Comment by zhujiammy** [comment]
First of all, I would like to thank bbonik for this project. I used the pythonnet library to build the emotion recognition function in the wpf project. Everything worked well, but due to the performance problem of pythonnet, I used python3.12.8 to retrain the model_emotion_pls=30_fullfeatures=False.joblib model, and tested the new model and the old model at the same time. I used all the pictures under data/images to compare the accuracy and found that the new model was more accurate. Then I converted the model to onnx and refactored it on c++. I compared the results of a lot of python and c++ runs, and now I have achieved good performance. My wpf project has removed the pythonnet library and used the c++ packaged dll. Now the performance has been very good. I have uploaded the improved project to github and hope it will be useful to you. Thank you! Thank you very much!

https://github.com/zhujiammy/facial-expression-analysis-main

**Comment by bbonik** [comment]
Thnx @zhujiammy for this update. That's really nice. It's good to have another updated option on this. Thnx for your contribution.
